src/stream/src/executor_v2/lookup.rs

```diff
@@ -122,40 +123,30 @@ impl ExecutorBuilder for LookupExecutorBuilder {
         let arrangement = params.input.remove(1);
         let stream = params.input.remove(0);
 
-        let arrangement_col_descs = arrangement
-            .schema()
-            .fields()
-            .iter()
-            .map(ColumnDesc::from_field_without_column_id)
-            .enumerate()
-            .map(|(idx, desc)| {
-                assert!(desc.field_descs.is_empty(), "sub-field not supported yet");
-                ColumnDesc {
-                    column_id: ColumnId::new(idx as i32),
-                    ..desc
-                }
-            })
-            .collect();
-
         let arrangement_order_rules = lookup
             .arrange_key
             .iter()
+            // TODO: allow descending order
             .map(|x| OrderPair::new(*x as usize, OrderType::Ascending))
             .collect();
 
-        assert_ne!(
-            lookup.arrange_fragment_id,
-            u32::MAX,
-            "arrange fragment id is not available"
-        );
-        let arrangement_keyspace_id =
-            unique_operator_id(lookup.arrange_fragment_id, lookup.arrange_operator_id);
+        let arrangement_table_id = match lookup.arrangement_table_id.as_ref().unwrap() {
+            ArrangementTableId::IndexId(x) => *x,
+            ArrangementTableId::TableId(x) => *x,
+        };
+
+        let arrangement_col_descs = lookup
+            .get_arrangement_table_info()?
```

Comment:
This is essentially from `order_desc` in `TableDesc`, which is arrange key + pk.
But the comment of `arrangement_col_descs` in `LookupExecutorParams` wrote
```
/// From the perspective of arrangements, `arrangement_col_descs` include all columns of the
/// `MaterializeExecutor`. For example, if we already have a table with 3 columns: `a, b,
/// _row_id`, and we create an arrangement with join key `a` on it. `arrangement_col_descs`
/// should contain all 3 columns.
```

Reply:
Nope, that's all columns 🤣 Index contains all columns, and its column desc also contains all columns.

Reply:
Oh sry, my mistake.